### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by akzhy**
Optionally add disqus by providing the value in `siteMetadata`

### CODE REVIEW

File: src/components/comments.js [hunk 1]
```diff
@@ -1,33 +1,36 @@
...
 function Comments({ data, title, location }) {
-  const disqusConfig = {
-    url: `${data.site.siteMetadata.siteUrl + location}`,
-    identifier: title,
-    title: title,
-  }
-  return (
-    <Disqus config={disqusConfig} />
-  )
+    if (data.site.siteMetadata.disqus === null) return <></>;
```
Comment (ahmad-reza619) [inline src/components/comments.js]: i would prefer returning a null here, but this is fine too

File: src/components/comments.js [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-            }
-        `}
-      render={data => <Comments data={data} title={title} location={location} />}
-    />
-  )
-}
\ No newline at end of file
+            `}
+            render={data => (
+                <Comments data={data} title={title} location={location} />
```
Comment (ahmad-reza619) [inline src/components/comments.js]: i think we can simplify the props here by only giving the url it needs.
something like this
```javascript
render={(data) => {
const url = data.site.siteMetadata.siteUrl + location;
return <Comments url={url} title={title} />
}}
```

File: src/components/comments.js [hunk 3]
```diff
@@ -1,33 +1,36 @@
-import React from 'react';
+import React from "react";
 
 import { StaticQuery, graphql } from "gatsby";
 
-import { Disqus } from 'gatsby-plugin-disqus'
+import { Disqus } from "gatsby-plugin-disqus";
 
 function Comments({ data, title, location }) {
```
Comment (ahmad-reza619) [inline src/components/comments.js]: and the props here changed with `url`

File: src/components/comments.js [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
-    identifier: title,
-    title: title,
-  }
-  return (
-    <Disqus config={disqusConfig} />
-  )
+    if (data.site.siteMetadata.disqus === null) return <></>;
+
+    const disqusConfig = {
+        url: `${data.site.siteMetadata.siteUrl + location}`,
```
Comment (ahmad-reza619) [inline src/components/comments.js]: and so, this would become 
```javascript
const disqusConfig = {
   url,
   ...other
}
```

**Review by ahmad-reza619 — COMMENTED** [review]
Yes thank you, i don't have many experience with gatsby so it's a bit hard for me😅 

By the way, i got some recommendation here, what do you think? I would gladly implement this myself if you're busy 😄

File: src/components/comments.js (hunk 1 above)
Comment (akzhy) [reply src/components/comments.js]: I haven't tested it, but I believe we need to return some components? (`React.Fragment` or `<></>` for components.

File: src/components/comments.js (hunk 2 above)
Comment (akzhy) [reply src/components/comments.js]: Yeah that would look more pretty I guess.

**Comment by akzhy** [comment]
Feel free to share any recommendations you have @ahmad-reza619,

**Comment by ahmad-reza619** [comment]
> Feel free to share any recommendations you have @ahmad-reza619,

will do, i'm glad to help. 😄

File: src/components/comments.js (hunk 1 above)
Comment (ahmad-reza619) [reply src/components/comments.js]: you can take a look at [this](https://stackoverflow.com/questions/42083181/is-it-possible-to-return-empty-in-react-render-function).

**Comment by ahmad-reza619** [comment]
how about we add some test too? so that it will guarantee our template stay safe and sound 😃

**Comment by ahmad-reza619** [comment]
hey @akzhy i've added some optimization. anyway, gatsby complains about disqus option that is not present in `gatsby-config.js` file. could we add some initial values for it?

**Comment by akzhy** [comment]
Hey @ahmad-reza619 is it a warning? If so where does it appear? There are no errors/warnings on my device

**Comment by ahmad-reza619** [comment]
Screen Shot 2020-03-15 at 21 51 20

it's an error unfortunately, it looks like this

**Comment by ahmad-reza619** [comment]
and why did you use yarn, when you used npm before? there is 2 lock file now, and from my experience this could cause issues

**Comment by akzhy** [comment]
Can you try clearing the cache ? `gatsby clean`.
For the lockfile part, I switched to yarn a couple of months ago 😅, forgot I used npm for this one.
@ahmad-reza619

**Comment by ahmad-reza619** [comment]
so, which one are we going to use?

**Comment by akzhy** [comment]
Yarn is pretty good

**Comment by akzhy** [comment]
Did clearing the cache work ? @ahmad-reza619

**Comment by ahmad-reza619** [comment]
nope, i was using node 8.6 so i tried to switch to node 13, but errors still happen, so now i tried with yarn

**Comment by ahmad-reza619** [comment]
all right it works on yarn with node v13

**Comment by akzhy** [comment]
Okay great we can proceed with yarn

**Comment by ahmad-reza619** [comment]
i've removed npm lock file so we should be good for now. and maybe we'll mention to use node v13 in readme perhaps?

**Comment by ahmad-reza619** [comment]
and let's not forget to mention about disqus in readme

**Comment by akzhy** [comment]
Yeah we will need to update the readme to inform about disqus

**Comment by akzhy** [comment]
Okay, looks good to me

**Review by ahmad-reza619 — APPROVED** [review]
all green. it's good to go @akzhy

**Pull request merged**